source/daplink/drag-n-drop/vfs_manager.c

```diff
@@ -615,12 +615,15 @@ static void transfer_update_file_info(vfs_file_t file, uint32_t start_sector, ui
 static void transfer_reset_file_info()
 {
     vfs_mngr_printf("vfs_manager transfer_reset_file_info()\r\n");
-    if (file_transfer_state.stream_open) {
-        transfer_update_state(ERROR_ERROR_DURING_TRANSFER);
-    } else {
-        file_transfer_state = default_transfer_state;
-        abort_remount();
-    }
+    //check if the data started streaming; size can be updated on matching start sector and stream type
+    if(file_transfer_state.stream_started && (TRASNFER_FINISHED != file_transfer_state.transfer_state)){
+        //file to program can change but all fields should be preserved
+        file_transfer_state.file_to_program = VFS_FILE_INVALID;
```

Comment:
You'll probably want to reset `file_size` and `start_sector` as well, since these may be set to the wrong value from the deleted file.

The value `start_sector` is a bit problematic, since the stream or root directory could have written this. One way to safely handle this is to create two variables, one for the start sector detected by the stream and one for the root directory - maybe something like `start_sector` and `file_start_sector`. Then you could reset `file_start_sector` here while leaving `start_sector` unchanged.  You could then make use of this as part of the optional finish check:
```C
    file_transfer_state.file_info_optional_finish =
        (file_transfer_state.file_to_program != VFS_FILE_INVALID) &&
        (file_transfer_state.size_transferred >= file_transfer_state.file_size) &&
        (file_transfer_state.file_size > 0) &&
        (file_transfer_state.start_sector == file_transfer_state.file_start_sector);
```